addons/website_google_map/static/src/js/website_google_map.js

```diff
@@ -81,6 +83,15 @@ function initialize_map() {
             set_marker(odoo_partner_data.partners[i]);
         }
         new MarkerClusterer(map, markers, options);
+
+        // auto center map and auto zoom
+        var bounds = new google.maps.LatLngBounds();
+        markers.forEach(function(n) {
+            bounds.extend(n.position);
+        });
+        map.setCenter(bounds.getCenter());
+        map.fitBounds(bounds);
+        map.setZoom(map.getZoom() - 1);
```

Comment:
Just so I understand (did not test, just reading the code): why not setting the right zoom at initialization?